### ISSUE TITLE
Panic after join when dataframe has a column of dtype pl.Object and resulting dataframe has only one row

### ISSUE THREAD

**Issue by ATL2001** [pola-rs/polars#19855]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

the following  code worked in polars 1.12 but panics in 1.13 and 1.14.  

```python
import polars as pl

print(pl.__version__)

df1 = pl.DataFrame(
    [
        ["10"],
        ["11"],
    ],
    schema={"name": pl.String},
    orient="row",
)

df2 = pl.DataFrame(
    [
        ["20", "random python object0, not a string"],
        # ["21", "another random python object"],
    ],
    schema={"name": pl.String, "ob": pl.Object},
    orient="row",
)

joined_df = df1.with_row_index("index").join(
    df2.with_row_index("index"),
    how="inner",
    on="index"
)
print(joined_df)
```
```python-traceback
1.14.0
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_17468\1518936687.py in <cell line: 0>()
     26     on="index"
     27 )
---> 28 print(joined_df)

~\myenv\Lib\site-packages\polars\dataframe\frame.py in __str__(self)
   1177 
   1178     def __str__(self) -> str:
-> 1179         return self._df.as_str()
   1180 
   1181     def __repr__(self) -> str:

PanicException: called `Result::unwrap()` on an `Err` value: SchemaMismatch(ErrString("expected object"))
```

### Issue description

it's not the in the join where the panic occurs, but it seems that doing anything with the resulting dataframe from the join causes the panic, including printing the dataframe.  What's more confusing to me is that when the resulting dataframe has more than one row it works fine (if you uncomment the second row in the data for df2 for example).

### Expected behavior

I'd expect a dataframe like this:
```python-traceback
shape: (1, 4)
┌───────┬──────┬────────────┬─────────────────────────────────┐
│ index ┆ name ┆ name_right ┆ ob                              │
│ ---   ┆ ---  ┆ ---        ┆ ---                             │
│ u32   ┆ str  ┆ str        ┆ object                          │
╞═══════╪══════╪════════════╪═════════════════════════════════╡
│ 0     ┆ 10   ┆ 20         ┆ random python object0, not a s… │
└───────┴──────┴────────────┴─────────────────────────────────┘
```

### Installed versions

<details>

```
--------Version info---------
Polars:              1.14.0
Index type:          UInt32
Platform:            Windows-10-10.0.19045-SP0
Python:              3.11.8 (main, Mar 22 2024, 13:25:41) [MSC v.1938 64 bit (AMD64)]
LTS CPU:             False

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
boto3                <not installed>
cloudpickle          2.2.1
connectorx           0.3.3
deltalake            <not installed>
fastexcel            <not installed>
fsspec               2023.10.0
gevent               <not installed>
google.auth          <not installed>
great_tables         <not installed>
matplotlib           3.6.3
nest_asyncio         1.6.0
numpy                1.24.3
openpyxl             3.0.10
pandas               2.0.2
pyarrow              15.0.0
pydantic             2.8.2
pyiceberg            <not installed>
sqlalchemy           1.4.51
torch                <not installed>
xlsx2csv             0.8.3
xlsxwriter           3.2.0
```

</details>

**Comment by samuelstevens** [comment]
I am having the same issue with 1.14.0 on linux.

<details><summary>Installed details</summary>

```
--------Version info---------
Polars:              1.14.0
Index type:          UInt32
Platform:            Linux-5.4.0-198-generic-x86_64-with-glibc2.31
Python:              3.11.9 (main, Aug 14 2024, 05:07:28) [Clang 18.1.8 ]
LTS CPU:             False

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               5.4.1
boto3                <not installed>
cloudpickle          3.1.0
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               2024.9.0
gevent               <not installed>
google.auth          <not installed>
great_tables         <not installed>
matplotlib           3.9.2
nest_asyncio         1.6.0
numpy                2.1.3
openpyxl             <not installed>
pandas               2.2.3
pyarrow              18.0.0
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                2.5.1+cu124
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

**Comment by ATL2001** [comment]
not sure what happened, or what fixed it, but I just checked in 1.17.1 and the issue is resolved.  Closing this issue :)
